### PR TITLE
Ptr ground station firmware preferences

### LINKED ISSUE #3: System preferences saving
Add a way to save preferences to prevent manual parameter setting on every reboot. Use json to store configuration file, same as in PTR_mega_firmware

### PR DESCRIPTION
**Pull request by Bdabrowsky**
Added automatic configuration saving via JSON file. Fixed minor bugs in GUI

### CODE REVIEW

File: src/GNSS.cpp [hunk 1]
```diff
@@ -4,6 +4,9 @@
 #include <MicroNMEA.h>
 #include "GNSS.h"
 
+
+#define pi 3.14159265358979323846
```
Comment (bartekM-pl) [inline src/GNSS.cpp]: Macro definitions - use capital letters

File: src/GNSS.cpp [hunk 2]
```diff
@@ -15,13 +18,19 @@ float myLon    = 0.0f;
...
 
     if (myGNSS.begin(Serial1) == false) {
         Serial.println(F("Ublox init Failed. Freezing."));
         //while (1);
+        return false;
     }
+    return true;
+}
+
+double toRad(double degree) {
```
Comment (bartekM-pl) [inline src/GNSS.cpp:30]: Is Double necessary?

File: src/GNSS.cpp [hunk 3]
```diff
@@ -41,13 +50,14 @@ void GNSS_srv(){
...
 float GNSS_calcDistance(float targetLat, float targetLon){
-    if(myFix == 0)
+   if(myFix == 0)
         return -1.0f;
-
-    float rlat1 = 3.14f * myLat / 180.0f;
-    float rlat2 = 3.14f * targetLat / 180.0f;
+    
+    /*
+    float rlat1 = 3.141592f * myLat / 180.0f;
```
Comment (bartekM-pl) [inline src/GNSS.cpp]: You have Pi as macro definition - use it

File: src/lora.cpp [hunk 4]
```diff
@@ -32,8 +34,13 @@ static uint16_t packetCounter[5] = {0,0,0,0,0};
 static float packet_rate = 0;
 static uint8_t LORA_newPacketReceivedOLED = 0;
 
+float LORA_currentFrequencyMHz = 434.25;
```
Comment (bartekM-pl) [inline src/lora.cpp]: ".f" missing

File: src/lora.cpp [hunk 5]
```diff
@@ -44,7 +51,7 @@ void LORA_init(){
         while (true);
     }
 
-    radio.setFrequency(433.0);
+    radio.setFrequency(433);
```
Comment (bartekM-pl) [inline src/lora.cpp]: Should be float

File: src/lora.cpp [hunk 6]
```diff
@@ -44,7 +51,7 @@ void LORA_init(){
...
     radio.setBandwidth(125);        // 7.8, 10.4, 15.6, 20.8, 31.25, 41.7, 62.5, 125, 250, 500
```
Comment (bartekM-pl) [inline src/lora.cpp:55]: Should be float

File: src/main.cpp [hunk 7]
```diff
@@ -63,110 +72,49 @@ void setup() {
...
-            //     snprintf(buf, sizeof(buf), "State %2i Bat %2.1fV", 4, lora_binmin_packet_d.vbat_10/10.0);
-            //     u8g2->drawStr(0, 51, buf);
-            //     snprintf(buf, sizeof(buf), "RSSI: %+.2f dbm", radio.getRSSI());
-            //     u8g2->drawStr(0, 64, buf);
-            //     u8g2->sendBuffer();
-            // }
-            
-           */
\ No newline at end of file
+  MAVLink_srv();
```
Comment (bartekM-pl) [inline src/main.cpp]: This should be on separated branch

**Review by bartekM-pl — CHANGES_REQUESTED** [review]
Check comments

File: src/GNSS.cpp (hunk 1 above)
Comment (Bdabrowsky) [reply src/GNSS.cpp]: Removed altogether in favour of builtin M_PI

File: src/GNSS.cpp (hunk 2 above)
Comment (Bdabrowsky) [reply src/GNSS.cpp:30]: Double yields better results with distance calculations

File: src/GNSS.cpp (hunk 3 above)
Comment (Bdabrowsky) [reply src/GNSS.cpp]: Removed this part of code

**Review by bartekM-pl — APPROVED** [review]

**Pull request merged**